tests/Resources/fs.addontest.js

```diff
@@ -0,0 +1,1493 @@
+			// (and this may be platform-specific behavior)
+			should.throws(() => {
+				fs.accessSync(thisFilePath, fs.constants.W_OK);
+			}, Error);
+			// TODO Verify the error.code value is EACCESS!
+		});
+
+		it('checks that this file is NOT executable properly', () => {
+			should.throws(() => {
+				fs.accessSync(thisFilePath, fs.constants.X_OK);
+			}, Error);
+			// TODO Verify the error.code value is EACCESS!
+		});
+	});
+
+	// TODO: #appendFile()
+
+	describe('#appendFileSync()', () => {
+		it('is a function', () => {
+			should(fs.appendFileSync).be.a.Function;
+		});
+	});
+
+	// TODO: #close()
+
+	describe('#closeSync()', () => {
+		it('is a function', () => {
+			should(fs.closeSync).be.a.Function;
+		});
+
+		it('returns undefined', () => {
+			const fd = fs.openSync(thisFilePath);
+			const result = fs.closeSync(fd);
+			should(result).be.undefined();
+		});
+	});
+
+	describe('#copyFile()', () => {
+		it('is a function', () => {
+			should(fs.copyFile).be.a.Function;
+		});
+
+		it('copies file asynchronously to destination', function (finished) {
+			this.slow(2000);
+			this.timeout(5000);
+
+			const dest = path.join(Ti.Filesystem.tempDirectory, 'fs.addontest.js');
+			// ensure file doesn't already exist
+			const destFile = Ti.Filesystem.getFile(dest);
+			if (destFile.exists()) {
+				should(destFile.deleteFile()).eql(true);
+			}
+			should(destFile.exists()).eql(false);
+
+			fs.copyFile(thisFilePath, dest, err => {
+				try {
+					should(err).not.be.ok;
+					fs.existsSync(dest).should.eql(true);
+					// TODO: Read in the file and compare contents? Check filesize matches?
+					finished();
+				} catch (e) {
+					finished(e);
+				}
+			});
+		});
+	});
+
+	describe('#copyFileSync()', () => {
+		it('is a function', () => {
+			should(fs.copyFileSync).be.a.Function;
+		});
+
+		it('copies file synchronously to destination', () => {
+			const dest = Ti.Filesystem.tempDirectory + 'fs.addontest.js';
+			// ensure file doesn't already exist
+			const destFile = Ti.Filesystem.getFile(dest);
+			if (destFile.exists()) {
+				should(destFile.deleteFile()).eql(true);
+			}
+			should(destFile.exists()).eql(false);
+
+			fs.copyFileSync(thisFilePath, dest);
+			fs.existsSync(dest).should.eql(true);
+			// TODO: Read in the file and compare contents? Check filesize matches?
+		});
+
+		// TODO: Check that we fail if file already exists with flag fs.constants.COPYFILE_EXCL
+		// TODO: Check that we overwrite byudefault if file already exists without flag
+	});
+
+	describe('#exists()', () => {
+		it('is a function', () => {
+			should(fs.exists).be.a.Function;
+		});
+
+		it('checks that this file exists properly', finished => {
+			fs.exists(thisFilePath, exists => {
+				try {
+					exists.should.eql(true);
+					finished();
+				} catch (e) {
+					finished(e);
+				}
+			});
+		});
+
+		it('checks that non-existent file returns false', finished => {
+			fs.exists('/some/made/up/path', exists => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/f...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/fs.addontest.js;
  Line: 244;
-->

- :warning: tests/Resources/fs.addontest.js line 244 – 'fs.exists' was deprecated since v4.0.0. Use 'fs.stat()' or 'fs.access()' instead. (node/no-deprecated-api)